File: src/oraculus_di_auditor/analysis/text_utils.py
"""Shared utility functions for analysis modules."""

from __future__ import annotations

from typing import Any


def extract_text_content(doc: dict[str, Any]) -> str:
    """Extract all text content from document for analysis.

    Args:
        doc: Normalized document dict

    Returns:
        Concatenated text content from all sections
    """
    text_parts = []

    # Extract from raw_text field
    if "raw_text" in doc and isinstance(doc["raw_text"], str):
        text_parts.append(doc["raw_text"])

    # Extract from sections
    sections = doc.get("sections", [])
    if isinstance(sections, list):
        for section in sections:
            if isinstance(section, dict) and "content" in section:
                text_parts.append(str(section["content"]))

    return " ".join(text_parts)

File: mobile/lib/analysis/textUtils.ts
/**
 * Shared utility functions for analysis modules.
 * 
 * Direct port of src/oraculus_di_auditor/analysis/text_utils.py
 */

import { NormalizedDocument } from './types';

/**
 * Extract all text content from document for analysis.
 * 
 * Mirrors Python: extract_text_content(doc)
 * Extracts from raw_text and sections[].content, joins with space.
 */
export function extractTextContent(doc: NormalizedDocument): string {
  const textParts: string[] = [];

  // Extract from raw_text field
  if ('raw_text' in doc && typeof doc.raw_text === 'string') {
    textParts.push(doc.raw_text);
  }

  // Extract from sections
  const sections = doc.sections;
  if (Array.isArray(sections)) {
    for (const section of sections) {
      if (
        section !== null &&
        typeof section === 'object' &&
        'content' in section
      ) {
        textParts.push(String(section.content));
      }
    }
  }

  return textParts.join(' ');
}
